src/da/celestia.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::HashMap;
+use std::fs;
+
+use async_trait::async_trait;
+use bollard::models::{HostConfig, Mount, PortBinding};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::app::config::AppChainConfig;
+use crate::cli::prompt::get_boolean_input;
+use crate::da::da_layers::{DaClient, DaError};
+use crate::utils::docker::{container_exists, kill_container, run_docker_image};
+use crate::utils::paths::get_celestia_home;
+use std::thread;
+use std::time::Duration;
+
+pub struct CelestiaClient;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CelestiaConfig {
+    pub ws_provides: String,
+    pub http_provider: String,
+    pub nid: String,
+    pub auth_token: String,
+    pub address: String,
+}
+
+#[derive(Error, Debug)]
+pub enum CelestiaError {
+    #[error("Faucet funds needed for DA to be submitted")]
+    FaucetFundsNeeded,
+    #[error("Celestia light node setup failed")]
+    SetupError,
+}
+
+const CELESTIA_DOCS: &str = "https://docs.celestia.org/developers/celestia-app-wallet#fund-a-wallet";
+
+#[async_trait]
+impl DaClient for CelestiaClient {
+    async fn generate_da_config(&self, config: &AppChainConfig) -> Result<(), DaError> {
+        let celestia_home = get_celestia_home().unwrap();
+        let file_keys_txt = celestia_home.join("keys.txt");
+        let file_auth_txt = celestia_home.join("auth.txt");
+
+        if !file_keys_txt.exists() || !file_auth_txt.exists() {
+            let run_cmd = vec![
+                "sh",
+                "-c",
+                "celestia light init --p2p.network=mocha > /home/celestia/keys.txt &&\
+                 celestia light auth admin --p2p.network=mocha > /home/celestia/auth.txt"
+            ];
+            run_celestia_light_node(run_cmd).await.unwrap();
+            // Waits for docker container to execute the commands and generate the keys
+            thread::sleep(Duration::from_secs(5));
```

Comment:
is there no other way to handle this

Reply:
So if we don't wait for docker to complete the command execution that we called on line 52 the `keys.txt` and `auth.txt` files aren't properly formed and is empty when we try to build `da_conf.json`. But yeah I think we can add some check to await if the container has exited or not because execution time will be system dependent and this is bad code, I was just hacking it to make it work, will update.